src/languageTools/LanguageClientWrapper.js

```diff
@@ -0,0 +1,377 @@
+            return $.Deferred().reject();
+        }
+
+        var requestHandler = this._onRequestHandler[params.type];
+        if(validateHandler(requestHandler)) {
+            return requestHandler.call(null, params.params);
+        }
+        console.log("No handler provided for server request type : ", params.type);
+        return $.Deferred().reject();
+
+    };
+
+    LanguageClientWrapper.prototype._onNotificationDelegator = function (params) {
+        if (!params || !params.type) {
+            console.log("Invalid Parameters provided for server notification type : ", params.type);
+        }
+
+        var notificationHandlers = this._onNotificationHandlers[params.type];
+        if(notificationHandlers && Array.isArray(notificationHandlers) && notificationHandlers.length) {
+            notificationHandlers.forEach(function (handler) {
+                if(validateHandler(handler)) {
+                    handler.call(null, params.params);
+                }
+            });
+        } else {
+            console.log("No handlers provided for server notification type : ", params.type);
+        }
+    };
+
+    LanguageClientWrapper.prototype._request = function (type, params) {
+        params = MessageHandler.constructRequestParams(type, params);
+        if (params) {
+            return this._requestClient(params);
+        }
+
+        console.log("Invalid Parameters provided for request type : ", type);
+        return $.Deferred().reject();
+    };
+
+    LanguageClientWrapper.prototype._notify = function (type, params) {
+        params = MessageHandler.constructNotificationParams(type, params);
+        if (params) {
+            this._notifyClient(params);
+        } else {
+            console.log("Invalid Parameters provided for notification type : ", type);
+        }
+    };
+
+    LanguageClientWrapper.prototype._addOnRequestHandler = function (type, handler) {
+        if (validateHandler(handler)) {
+            this._onRequestHandler[type] = handler;
+        }
+    };
+
+    LanguageClientWrapper.prototype._addOnNotificationHandler = function (type, handler) {
+        if (validateHandler(handler)) {
+            if (!this._onNotificationHandlers[type]) {
+                this._onNotificationHandlers[type] = [];
+            }
+
+            this._onNotificationHandlers[type].push(handler);
+        }
+    };
+
+    /**
+        Requests
+    */
+    //start
+    LanguageClientWrapper.prototype.start = function (params) {
+        params = MessageHandler.constructRequestParams(ToolingInfo.LANGUAGE_SERVICE.START, params);
+        return this._startClient(params);
+    };
+
+    //shutdown
+    LanguageClientWrapper.prototype.stop = function () {
+        return this._stopClient();
+    };
+
+    /**
+        textDocument requests
+    */
+    //completion
+    LanguageClientWrapper.prototype.requestHints = function (params) {
+        return this._request(ToolingInfo.FEATURES.CODE_HINTS, params);
+    };
+
```

Comment:
Add wrapper method to resolveHints(method: ‘completionItem/resolve’) also.
https://microsoft.github.io/language-server-protocol/specification#completionItem_resolve

Reply:
ResolveHints will be called from Provider insertHints function.

Reply:
ignore this